superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FilterScope/utils.test.ts

```diff
@@ -0,0 +1,18204 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Layout } from 'src/dashboard/types';
+import { buildTree } from './utils';
+
+// The types defined for Layout and sub elements is not compatible with the data we get back fro a real dashboard layout
+// This test file is using data from a real example dashboard to test real world data sets.  ts-ignore is set for this entire file
+// until we can reconcile adjusting types to match the actual data structures used
+
+describe('Ensure buildTree does not throw runtime errors when encountering an invalid node', () => {
+  const node = {
+    children: ['TABS-97PVJa11D_'],
+    id: 'ROOT_ID',
+    type: 'ROOT',
+    parents: [],
+  };
+
+  const treeItem = {
+    children: [],
+    key: 'ROOT_ID',
+    type: 'ROOT',
+    title: 'All panels',
+  };
+
+  const layout: Layout = {
```

Comment:
Could/should this be moved to a separate fixtures type of file to make the test more readable?

Reply:
@lyndsiWilliams that is a great point.  I am waiting to hear back on timing as this is desired for a patch asap.  Depending on time I will either make change in this PR, or I will do a small refinement follow on with the change so we are not blocked waiting on the git checks to re-run.

Reply:
Good idea, @lyndsiWilliams. I'm going to approve and merge so that we can get this bug fix in, and @eric-briscoe will follow up with a new PR to clean up the test.